Packages/ChineseNumber/Sources/ChineseNumber/suzhou_number.m

```diff
@@ -0,0 +1,111 @@
+#include <string.h>
+#include <stdlib.h>
+#include "trim_zero.h"
+
+#define BUFFER_SIZE 80
+
+struct suzhou_number_struct
+{
+    const char *int_part;
+    size_t int_trimmed_zero_count;
+    const char *dec_part;
+};
+
+static struct suzhou_number_struct build_struct(const char *int_part, const char *dec_part) {
+    const char *initial_int_part = trim_zero (int_part, true);
+    const char *initial_dec_part = trim_zero (dec_part, false);
+    const char *trimmed_int_part = trim_zero (initial_int_part, false);
+    size_t int_trimmed_zero_count = 0;
+    if (!strlen (initial_dec_part)) {
+        int_trimmed_zero_count = strlen (initial_int_part) - strlen (trimmed_int_part);
+    } else {
+        free ((void *) trimmed_int_part);
+        trimmed_int_part = initial_int_part;
+    }
+
+    if (!strlen (trimmed_int_part)) {
+        memcpy((void *) trimmed_int_part, (void *) "0", 2);
```

Comment:
This is only "safe" because from the implementation detail we know that `trim_zero` always allocates a block of size `BUFFER_SIZE`, but normally this `strlen`-`memcpy` operation is best avoided…

Reply:
The library is now in Swift :)